pkg/apis/pipeline/v1beta1/pipelinerun_types.go

```diff
@@ -449,6 +487,12 @@ const (
 	GracefullyStoppedSkip SkippingReason = "PipelineRun was gracefully stopped"
 	// MissingResultsSkip means the task was skipped because it's missing necessary results
 	MissingResultsSkip SkippingReason = "Results were missing"
+	// PipelineTimedOutSkip means the task was skipped because the PipelineRun has passed its overall timeout.
+	PipelineTimedOutSkip SkippingReason = "PipelineRun timeout has been reached"
+	// TasksTimedOutSkip means the task was skipped because the PipelineRun has passed its Timeouts.Tasks.
+	TasksTimedOutSkip SkippingReason = "PipelineRun Tasks timeout has been reached"
+	// FinallyTimedOutSkip means the task was skipped because the PipelineRun has passed its Timeouts.Finally.
+	FinallyTimedOutSkip SkippingReason = "PipelineRun Finally timeout has been reached"
```

Comment:
How is this possible with the constraint we have [documented](https://github.com/tektoncd/pipeline/blob/main/docs/pipelineruns.md#configuring-a-failure-timeout)?

```
timeouts.pipeline >= timeouts.tasks + timeouts.finally
```

Does this reason only apply to `finally` tasks? or can also be applied to `dag tasks`?

Reply:
There are a few ways - dag tasks finishing quickly enough that the remaining time until the pipeline timeout is greater than the finally timeout, no pipeline timeout at all, etc. We’re using a different skipping reason for finally vs dag tasks to make the specific timeout setting that caused the timeout clear.